### ISSUE TITLE
PointerEventConverter.expand: The getter 'pointer' was called on null.

### ISSUE THREAD

**Issue by cvolzke4** [flutter/flutter#39031]
```
From crash:

NoSuchMethodError: The getter 'pointer' was called on null.
#0      PointerEventConverter.expand (package:flutter/src/gestures/converter.dart:254)
#1      _SyncIterator.moveNext (dart:core-patch/core_patch.dart:144)
#2      ListQueue.addAll (dart:collection/queue.dart:655)
#3      _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:84)
#4      _rootRunUnary (dart:async/zone.dart:1136)
#5      _CustomZone.runUnary (dart:async/zone.dart:1029)
#6      _CustomZone.runUnaryGuarded (dart:async/zone.dart:931)
#7      _invoke1 (dart:ui/hooks.dart:252)
#8      _dispatchPointerDataPacket (dart:ui/hooks.dart:161)
```

**Comment by BondarenkoStas** [comment]
@cvolzke4 Please provide ```flutter doctor -v``` output as well as the steps to reproduce the issue.

**Comment by keshava19** [comment]
```
NoSuchMethodError: NoSuchMethodError: The getter 'pointer' was called on null.
Receiver: null
Tried calling: pointer
  File "converter.dart", line 254, in PointerEventConverter.expand
  File "core_patch.dart", line 152, in _SyncIterator.moveNext
  File "queue.dart", line 655, in ListQueue.addAll
  File "binding.dart", line 84, in _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket
  File "zone.dart", line 1136, in _rootRunUnary
  File "zone.dart", line 1029, in _CustomZone.runUnary
  File "zone.dart", line 931, in _CustomZone.runUnaryGuarded
  File "hooks.dart", line 250, in _invoke1
  File "hooks.dart", line 159, in _dispatchPointerDataPacket
```
Flutter doctor :-
```console
keshava@debian:~$ flutter doctor -v
[✓] Flutter (Channel stable, v1.7.8+hotfix.4, on Linux, locale en_IN.UTF-8)
    • Flutter version 1.7.8+hotfix.4 at /home/keshava/Development/flutter
    • Framework revision 20e59316b8 (8 weeks ago), 2019-07-18 20:04:33 -0700
    • Engine revision fee001c93f
    • Dart version 2.4.0

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /home/keshava/Android/Sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /home/keshava/Development/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[!] Android Studio (version 3.5)
    • Android Studio at /home/keshava/Development/android-studio
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.38.0)
    • VS Code at /usr/share/code
    • Flutter extension version 3.4.1

[✓] Connected device (1 available)
    • Redmi 6 Pro • 1f04e0bb0405 • android-arm64 • Android 9 (API 28)

! Doctor found issues in 1 category.
```

**Comment by cvolzke4** [comment]
@BondarenkoStas I don't know how to reproduce this.

**Comment by BondarenkoStas** [comment]
@keshava19 Maybe you can come up with any steps how to reproduce it? When it occurs? What command do you run?

**Comment by keshava19** [comment]
@BondarenkoStas I too don't know how to reproduce this bug.

**Comment by BondarenkoStas** [comment]
I would advise you to post it to SO as well. It will be very hard to investigate without knowing where to start, but maybe somebody from community already encountered the issue and can help with it. In this case please paste the link to the question here.

**Comment by keshava19** [comment]
```console
I/flutter (24943): 'package:flutter/src/gestures/converter.dart': Failed assertion: line 249 pos 20: '_pointers.containsKey(datum.device)': is not true. 
I/flutter (24943): #0 _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:40:39) 
I/flutter (24943): #1 _AssertionError._throwNew (dart:core-patch/errors_patch.dart:36:5) 
I/flutter (24943): #2 PointerEventConverter.expand 
package:flutter/…/gestures/converter.dart:249
I/flutter (24943): #3 _SyncIterator.moveNext (dart:core-patch/core_patch.dart:152:12) 
I/flutter (24943): #4 ListQueue.addAll (dart:collection/queue.dart:655:25) 
I/flutter (24943): #5 _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket 
package:flutter/…/gestures/binding.dart:84
I/flutter (24943): #6 _rootRunUnary (dart:async/zone.dart:1136:13) 
I/flutter (24943): #7 _CustomZone.runUnary (dart:async/zone.dart:1029:19) 
I/flutter (24943): #8 _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7) 
I/flutter (24943): #9 _invoke1 (dart:ui/hooks.dart:250:10) 
I/flutter (24943): #10 _dispatchPointerDataPacket (dart:ui/hooks.dart:159:5)
```

I got this when started the app in debug mode.

**Comment by crazecoder** [comment]
I also encountered this bug in the production environment, but because it is a third-party plug-in collection report,So I don't know where this error is caused. This is the reported log
```
1 #0 PointerEventConverter.expand (package:flutter/src/gestures/converter.dart:254)
--
2 #1 _SyncIterator.moveNext (dart:core-patch/core_patch.dart:144)
3 #2 ListQueue.addAll (dart:collection/queue.dart:655)
4 #3 GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:84)
5 #4 _rootRunUnary (dart:async/zone.dart:1136)
6 #5 _CustomZone.runUnary (dart:async/zone.dart:1029)
7 #6 _CustomZone.runUnaryGuarded (dart:async/zone.dart:931)
8 #7 _invoke1 (dart:ui/hooks.dart:263)
9 #8 _dispatchPointerDataPacket (dart:ui/hooks.dart:172)
```
```
1 #0 PointerEventConverter.expand (package:flutter/src/gestures/converter.dart:281)
--
2 #1 _SyncIterator.moveNext (dart:core-patch/core_patch.dart:144)
3 #2 ListQueue.addAll (dart:collection/queue.dart:655)
4 #3 GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:84)
5 #4 _rootRunUnary (dart:async/zone.dart:1136)
6 #5 _CustomZone.runUnary (dart:async/zone.dart:1029)
7 #6 _CustomZone.runUnaryGuarded (dart:async/zone.dart:931)
8 #7 _invoke1 (dart:ui/hooks.dart:263)
9 #8 _dispatchPointerDataPacket (dart:ui/hooks.dart:172)
```

**Comment by qq326646683** [comment]
same issue
<img width="830" alt="android_error2" src="https://user-images.githubusercontent.com/15717688/67460472-aa449400-f66d-11e9-93b7-0472110c3176.png">

**Comment by brorhb** [comment]
I have also encountered two times in production now. Sorry to say this is all the crash info I have right now 

![IMG_0015](https://user-images.githubusercontent.com/12571304/68199392-3fc11b80-ffbe-11e9-805d-6ed1bb782810.jpg)

**Comment by ogrenichv** [comment]
Any news? Was it fixed in 1.12?

**Comment by iapicca** [comment]
Hi @cvolzke4 
could you please provide a minimal reproducible code sample?
Thank you

**Comment by cvolzke4** [comment]
I can't reproduce this, I only have the stack trace reported from the app (#quill).

crazecoder mentioned this may reproduce the issue:
https://stackoverflow.com/q/55022915

**Comment by iapicca** [comment]
Hi @ogrenichv @keshava19 @brorhb @qq326646683 @crazecoder 
could you please provide a reproducible code sample?
Thank you

**Comment by mivanzhang** [comment]
similar question with flutter 1.9.1

**Comment by mivanzhang** [comment]
![image](https://user-images.githubusercontent.com/8842825/81521039-9dbd2a00-9378-11ea-838a-a69d38f0724b.png)
![image](https://user-images.githubusercontent.com/8842825/81521407-bda11d80-9379-11ea-84be-98bacc47d5d6.png)

**Comment by mivanzhang** [comment]
I found this bug has been fixed in https://github.com/flutter/flutter/commit/314eb3bae37f9daf68a3bd9a2d8c870c7771e6b6 and https://github.com/flutter/engine/commit/132d38cd222a9e9009d969defbdde658ad3079af
if you update flutter to 1.12,this will be fixed ,if not ,I advice you to cherry-pick both commits

**Comment by iapicca** [comment]
Hi all
for the issue to be workable is [needed](https://github.com/flutter/flutter/issues/39031#issuecomment-627878905) a minimal reproducible code sample,
could you please check if the issue is fixed with the latest stable version
Thank you

**Comment by mivanzhang** [comment]
Truely ,I don`t know how to reproduce the bug. We got lots of crashes with this bug ,almost 1% users will come into this situation

**Comment by lionoggo** [comment]
On Android devices, some special gestures may cause this problem, such as dragging three fingers on the screen to realize the screenshot  or other.

**Comment by chunhtai** [comment]
This should be fixed with the latest stable 1.17. Includes the tree finger gesture @lionoggo mentioned. I will close this issue for now. If you are still seeing this issue after upgrade to latest stable, please reopen this issue

**Comment by mivanzhang** [comment]
@chunhtai Is this your fix commit? I found this in flutter 1.12.
https://github.com/flutter/flutter/commit/314eb3bae37f9daf68a3bd9a2d8c870c7771e6b6 and https://github.com/flutter/engine/commit/132d38cd222a9e9009d969defbdde658ad3079af

**Comment by chunhtai** [comment]
@mivanzhang  Yes, it is
